utils/fixed_decimal/src/decimal.rs

```diff
@@ -552,6 +552,197 @@ impl FixedDecimal {
         self.check_invariants();
     }
 
+    /// Increments the digits by 1. if the digits are empty, it will add
+    /// an element with value 1 and assign the value of the upper magnitude to
+    /// the magnitude.
+    fn increment_abs_by_one(&mut self) {
+        for i in (0..self.digits.len()).rev() {
+            self.digits[i] += 1;
+            if self.digits[i] < 10 {
+                return;
+            }
+            self.digits[i] = 0;
```

Comment:
Nit: It would be nice if this function always left the digits list in a valid state when it returns (no trailing zeros). Since you know how many 0s you are setting, you should be able to truncate the array before returning out of the function.

Reply:
okay, now I use pop for the zeros.